registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/utils/NotificationUtility.java

```diff
@@ -136,20 +152,64 @@ public void sendNotification(RegistrationAdditionalInfoDTO registrationAdditiona
 				if (notificationType.equalsIgnoreCase("EMAIL")
 						&& (registrationAdditionalInfoDTO.getEmail() != null
 						&& !registrationAdditionalInfoDTO.getEmail().isEmpty())) {
-					sendEmailNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description);
+					sendEmailNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description,preferredLanguage);
 				} else if (notificationType.equalsIgnoreCase("SMS") && (registrationAdditionalInfoDTO.getPhone() != null
 						&& !registrationAdditionalInfoDTO.getPhone().isEmpty())) {
-					sendSMSNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description);
+					sendSMSNotification(registrationAdditionalInfoDTO, messageSenderDTO, attributes, description,preferredLanguage);
 				}
 			}
 		}
+		}
+	}
+
+	private List<String> getPreferredLanguages(InternalRegistrationStatusDto registrationStatusDto) throws ApisResourceAccessException, 
+	PacketManagerException, JsonProcessingException, IOException, JSONException {
+		String preferredLang=packetManagerService.getField(registrationStatusDto.getRegistrationId(), MappingJsonConstants.PREFERRED_LANGUAGE,
+				registrationStatusDto.getRegistrationType(), ProviderStageName.PACKET_VALIDATOR);
+		if(preferredLang!=null && !preferredLang.isBlank()) {
+			return List.of(preferredLang.split(","));
+		}else {
+			if(defaultTemplateLanguages!=null && !defaultTemplateLanguages.isEmpty()) {
+				return defaultTemplateLanguages;
+			}else {
+				List<Field> fields=List.of(MappingJsonConstants.class.getDeclaredFields());
+				List<String> jsonFields =new ArrayList<>();
+				fields.forEach(x ->jsonFields.add(x.getName()));
+				Map<String, String> idObjectMap=packetManagerService.getFields(registrationStatusDto.getRegistrationId(), jsonFields, 

```

Comment:
u r trying to get the field by mapping json key, it may not work if the mapping json value is different. For example - for dob key value could be dateOfBirth, Since u will be searching by dob key u may not get result, U should get mapping json value from json file

Reply:
I have added a new method in packet manager service to get all available id object fields.